### PR TITLE
fix: swiftLint --fix as pre-commit - WPB-2534

### PR DESCRIPTION
**Pull request by agisilaos**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-2534" title="WPB-2534" target="_blank">StoryWPB-2534</a>  Automate fixing SwiftLint warnings through autocorrect command
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

With this PR we fix a long standing problem where we would identify and see SwiftLint warnings after committing code. That would lead to many comments in PR's to fix such warnings. 

Now we have a script that runs before pushing any code to the repo. 

----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: .githooks/pre-push [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see http://www.gnu.org/licenses/.
+#
+
+#!/bin/bash
+echo "pre-commit started"
```
Comment (netbe) [inline .githooks/pre-push]:
```suggestion
echo "pre-push started"
```

File: .githooks/pre-push [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            fi
+        fi
+}
+
+if [[ -e "${SWIFT_LINT}" ]]; then
+    echo "SwiftLint version: $(${SWIFT_LINT} version)"
+    # Run only if not merging
+    if ! git rev-parse -q --verify MERGE_HEAD; then 
+        # Run for both staged and unstaged files
+        echo "Enter merging"
```
Comment (netbe) [inline .githooks/pre-push]: I don't really get it, if this is pre-push, why do we check about merge?

Comment (agisilaos) [reply .githooks/pre-push]: Hmm. 🤔 Maybe I left it there as I was messing around with the scripts. Do you think we should get rid of it?

File: wire-ios/Wire-iOS.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -8352,7 +8352,7 @@
 			);
 			runOnlyForDeploymentPostprocessing = 0;
 			shellPath = /bin/sh;
-			shellScript = "# Adds support for Apple Silicon brew directory\nexport PATH=\"$PATH:/opt/homebrew/bin\"\n \nif [ -z \"$CI\" ]; then\n  if which swiftlint >/dev/null; then\n    swiftlint\n  else\n    echo \"warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint\"\n  fi\nelse\n  echo \"Skipping SwiftLint in CI environment\"\nfi\n";
+			shellScript = "# Adds support for Apple Silicon brew directory\nexport PATH=\"$PATH:/opt/homebrew/bin\"\n \nif [ -z \"$CI\" ]; then\n  if which swiftlint >/dev/null; then\n    swiftlint --config ../.swiftlint.yml\n  else\n    echo \"warning: SwiftLint not installed, download from https://github.com/realm/SwiftLint\"\n  fi\nelse\n  echo \"Skipping SwiftLint in CI environment\"\nfi\n";
```
Comment (netbe) [inline wire-ios/Wire-iOS.xcodeproj/project.pbxproj]: what about moving all this duplicate script to a single bash script under `scripts/`

File: .githooks/pre-commit [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+end
+
+def running_on_ci 
+	!ENV['CI'].nil?
+end
+
+def swiftlint_exists
+	`which swiftlint`
+	$?.success?
+end
```
Comment (netbe) [inline .githooks/pre-commit:60]: need to uniform indentation on the file

File: scripts/githooks-install.sh [hunk 5]
```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
```
Comment (netbe) [inline scripts/githooks-install.sh:1]: can you add this file to the Xcode project to ease the editing of it?

File: scripts/run-swiftlint.sh [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# Adds support for Apple Silicon brew directory
```
Comment (netbe) [inline scripts/run-swiftlint.sh:1]: same here

**Pull request merged**